tests/integration/thrift_tests.py

```diff
@@ -406,6 +433,80 @@ def example(self, context):
 
         assert handler.edge_context == FakeEdgeContextFactory.DECODED_CONTEXT
 
+    def test_budget_header_pool_timeout(self):
+        """Test that the budget header is set in the headers with the pool timeout."""
+        retry_timeout_seconds = 100.0
+
+        class Handler(TestService.Iface):
+            def example(self, context):
+                self.context = context
+                return True
+
+        handler = Handler()
+
+        span_observer = mock.Mock(spec=SpanObserver)
+        with serve_thrift(handler, TestService) as server:
+            with baseplate_thrift_client(
+                server.endpoint, TestService, span_observer, timeout="1 second"
+            ) as context:
+                with context.example_service.retrying(
+                    attempts=3, budget=retry_timeout_seconds
+                ) as svc:
+                    svc.example()
+
+        assert handler.context.headers.get(b"Deadline-Budget").decode() == str(1000)
```

Comment:
nit: maybe add a comment to explain that this `1000` is coming from `"1 second"` above.